src/test/java/io/kokuwa/maven/helm/UploadMojoTest.java

```diff
@@ -387,4 +440,59 @@ private void assertCatalog(UploadMojo mojo, Path archive, String uploadUrl) {
 		}
 
 	}
+
+	private void assertUploadVerifySuccess(UploadMojo mojo, RequestMethod method, String path) {
+		mockHelmShowChart();
+		assertDoesNotThrow(() -> mojo.execute(), "upload failed");
+		verifyRequests(mojo, method, path);
+	}
+
+	private void assertUploadVerifyFail(UploadMojo mojo, RequestMethod method, String path) {
+		mockHelmShowChart();
+		assertThrows(MojoExecutionException.class, mojo::execute, "could not verify");
+		verifyRequests(mojo, method, path);
+	}
+
+	private void mockHelmShowChart() {
+		mock.stubFor(WireMock.put(WireMock.urlMatching(".*"))
+				.willReturn(WireMock.ok()
+						.withStatus(200)));
+		mock.stubFor(WireMock.get(WireMock.urlMatching(".*/index\\.yaml$"))
+				.willReturn(WireMock.ok()
+						.withStatus(200)
+						.withHeader("Content-Type", "application/yaml")
+						.withBody(getIndexYamlBody())));
+		mock.stubFor(WireMock.get(WireMock.urlMatching(".*\\.tgz$"))
+				.willReturn(WireMock.ok()
+						.withStatus(200)
+						.withHeader("Content-Type", "application/gzip")
+						.withBodyFile("app-0.1.0.tgz")));
+	}
+
+	private void verifyRequests(UploadMojo mojo, RequestMethod method, String path) {
+		List<LoggedRequest> requests = mock.findAll(RequestPatternBuilder.allRequests());
+		LoggedRequest request = requests.get(0);
+		assertEquals(method, request.getMethod(), "method");
+		assertEquals((mojo.getUploadRepoStable().getUrl().startsWith("https")
+				? "https://127.0.0.1:" + mock.getHttpsPort()
+				: "http://127.0.0.1:" + mock.getPort()) + path, request.getAbsoluteUrl(), "url");
+		assertEquals("application/gzip", request.getHeader(HttpHeaders.CONTENT_TYPE), "content-type");
+	}
+
+	private String getIndexYamlBody() {
+		String indexYamlBody = "apiVersion: v1\n" +
+				"entries:\n" +
+				"  app:\n" +
+				"    - created: 2023-11-05T12:15:23.451853285-06:00\n" +
+				"      description: Dummy chart for testing\n" +
+				"      digest: digesthash1\n" +
+				"      home: https://helm.sh/helm\n" +
+				"      name: app\n" +
+				"      sources:\n" +
+				"        - https://github.com/helm/helm\n" +
+				"      urls:\n" +
+				"        - " + "http://127.0.0.1:" + mock.getPort() + "/nexus/app-0.1.0.tgz" + "\n" +
+				"      version: 0.1.0";
+		return indexYamlBody;
+	}
```

Comment:
had to move the mock index.yaml file to inside of the class due to the dynamic port allocation and to bypass any issues with having to clean up an actual index.yaml file after a failed test